pkg/controllers/terminating/v1alpha1/controller.go

```diff
@@ -65,16 +66,27 @@ func NewController(kubeClient client.Client, coreV1Client corev1.CoreV1Interface
 
 // Reconcile executes a reallocation control loop for the resource
 func (c *Controller) Reconcile(ctx context.Context, object client.Object) (reconcile.Result, error) {
+	n := object.(*v1.Node)
+	if n.DeletionTimestamp == nil || !node.IsKarpenterProvisioned(n) {
```

Comment:
node.HasTerminationFinalizer?

Reply:
Thanks. We want to continue this reconcile loop until the finalizer is removed. Replacing the line with 
`if !functional.ContainsString(n.Finalizers, provisioning.KarpenterFinalizer) {`